bundles/org.openhab.binding.comfoair/src/main/java/org/openhab/binding/comfoair/internal/datatypes/DataTypeBoolean.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.comfoair.internal.datatypes;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.comfoair.internal.ComfoAirCommandType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class to handle boolean values which are handled as decimal 0/1 states
+ *
+ * @author Holger Hees - Initial Contribution
+ * @author Hans Böhm - Refactoring
+ */
+public class DataTypeBoolean implements ComfoAirDataType {
```

Comment:
Please add `@NonNullByDefault`

Reply:
I left this out intentionally, since adding this would mean a lot of adoptions throughout the code. `convertFromState` method would have to be changed from `int[]` to `Integer[]`, since `int` is not nullable, with according changes to the referring classes. Beyond that I really don't know if annotations like `@Nullable Integer @Nullable []` are an improvement for the code in terms of readability.
So if not mandatory, I would like to avoid touching working code as far as possible. However, if you think that this would valuably improve the code quality, I will try my best to add these annotations.

Reply:
Arrays are a little different, using `int @Nullable []` should work. But yes, I agree that it is ugly. If you want you can use `@NonNullByDefault({})` on those array fields so that they can be ignored.

Reply:
I decided to stick to `int @Nullable []`. In the end it was less work than expected and I have to admit, that the annotations already served their purpose :wink: